pip_audit/_dependency_source/requirement.py

```diff
@@ -122,7 +133,19 @@ def collect(self) -> Iterator[Dependency]:
             for t in tmp_files:
                 t.unlink()
 
-    def _collect_from_files(self, filenames: list[Path]) -> Iterator[Dependency]:
+    def _parse_requirements_files(self, filenames: list[Path]):
+        # Parse requirements files without install in temporary virtual env.
+        parsed_packages = []
+        for filename in filenames:
+            requirements_file = RequirementsFile.from_file(filename)
+            requirements_dict = requirements_file.to_dict()["requirements"]
+            parsed_packages += [
+                (r["name"], r["specifier"][0].split("==")[1]) for r in requirements_dict
+            ]
+        for name, version in parsed_packages:
+            yield ResolvedDependency(name=name, version=Version(version))
```

Comment:
This needs to be hardened against URL-style requirements, parser errors, etc.

My suggestion would be to look at what was removed in #540, and reuse that to the greatest extent possible: it had a lot of "scar tissue" from different bugs reported over time.